mapedit/chunklst.cc

```diff
@@ -728,6 +753,12 @@ Chunk_chooser::Chunk_chooser(
 	g_signal_connect(G_OBJECT(chunk_adj), "value-changed",
 	                 G_CALLBACK(scrolled), this);
 	gtk_widget_show(vscroll);
+	// Scroll events.
+	GtkEventController *scroll_ctlr = GTK_EVENT_CONTROLLER(
```

Comment:
Hm. I wonder if scroll_ctlr and hbox1 below need to be added to the class and be destroyed when the Chunk_chooser is deleted. Most of the others are handled by gtk_box_pack_start or set_widget, but I wonder if they also need something like a call to g_object_unref to avoid leaking.

Reply:
You are right about `scroll_ctlr` needing a `g_object_unref`. It is not a `GtkWidget` and it is not created `floating`.
I have fully factorized the scroller in the browsers into the class `Object_browser` which is the parent class to all browsers, and I have put the `g_object_unref` into `Object_browser`'s destructor.

On the other hand, I do not see why you single `hbox1`, as it is also `gtk_box_pack_started` into `vbox`. I have written a quick check as putting it into `Object_browser` and checking its state in `Object_browser`'s destructor : it is destroyed when the destructor is invoked.